fix: Write poetryOptions to pyproject.toml

lmarsden opened the pull request:
* The poetryOptions object in the PythonProject was previously ignored when synthesizing pyproject.toml.
* poetryOptions declarations are now written to pyproject.toml.
* This allows specification of e.g. private PyPI registries via `[tool.poetry.source]` entries:

```
poetryOptions: {
    source: [
      {
        name: "myprivateregistry",
        url: "https://pkgs.dev.azure.com/myorg/_packaging/myprivateregistry/pypi/simple/",
        secondary: true,
      },
    ]
  },
```

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Chriscbr reviewed (changes requested):
Can you add some tests to `python-project.test.ts` or `poetry.test.ts` that illustrate the change?

lmarsden commented:
> Can you add some tests to `python-project.test.ts` or `poetry.test.ts` that illustrate the change?

Have now updated the code in `poetry.test.ts` to test for correct behaviour of the `poetry` options that were previously missing. `PoetryPyrojectOptions` (via `PoetryPyrojectOptionsWithoutDeps`) class now updated to support all `poetry` options documented at  https://python-poetry.org/docs/pyproject

Chriscbr commented on src/__tests__/python/poetry.test.ts:
```diff
@@ -14,15 +14,73 @@ test('poetry enabled', () => {
+  expect(snapshot['pyproject.toml']).toContain('url = "https://pypi.org/simple/"');
+  expect(snapshot['pyproject.toml']).toContain('[tool.poetry.scripts]');
+  expect(snapshot['pyproject.toml']).toContain('test-python-cli = "test-python-project.cli:cli"');
+  expect(snapshot['pyproject.toml']).toContain('[tool.poetry.extras]');
+  expect(snapshot['pyproject.toml']).toContain('cli = [ "package1", "package2" ]');
+  expect(snapshot['pyproject.toml']).toContain('[tool.poetry.plugins."blogtool.parsers"]');
+  expect(snapshot['pyproject.toml']).toContain('".rst" = "some_module:SomeClass"');
+  expect(snapshot['pyproject.toml']).toContain('[tool.poetry.urls]');
+  expect(snapshot['pyproject.toml']).toContain('"bug tracker" = "https://github.com/test-python-project/issues"');
```
This is getting pretty long and detailed - let's replace all of this (in the new test case) with

```js
expect(snapshot['pyproject.toml']).toMatchSnapshot();
```

lmarsden replied:
Thanks for the suggestion. Snapshot added and tests updated.

Chriscbr reviewed (changes requested):
Looks great @lmarsden, just a few small comments.

lmarsden commented on src/__tests__/python/poetry.test.ts:
```diff
@@ -14,15 +14,73 @@ test('poetry enabled', () => {
     classifiers: [
       'Development Status :: 4 - Beta',
     ],
+    poetryOptions: {
```
Done. Also see comment below.

lmarsden commented:
Looks like the build is failing at the `Commit and push changes (if changed)` step in `build.yml` as it's attempting to push some changes to my fork at https://github.com/lmarsden/projen

Are there any changes I need to make?

Chriscbr commented:
Looks like some kind of transient snapshot error because of some issues with our integration tests. Usually re-running `yarn build` fixes this
